### PR TITLE
Communication object

### PR DESCRIPTION
**Pull request by lazypower**
@mbruzek I backported your patches from #23  in this branch as well as adding my own tweaks to the registration code + object.

Some interesting findings if you can take a look I'd appreciate the validation. I ran this against v0.15.0 RC and passed the standup test. No additional validations have been done at this time, and additional refactoring may be required.

### CODE REVIEW

File: hooks/hooks.py [hunk 1]
```diff
@@ -189,50 +189,27 @@ def register_machine(apiserver, retry=False):
...
-
-    # print("Registration request %s" % request)
-    conn = httplib.HTTPConnection(parsed.hostname, parsed.port)
-    conn.request("POST", "/api/v1beta1/minions", json.dumps(request), headers)
-
-    response = conn.getresponse()
-    body = response.read()
-    print(body)
-    result = json.loads(body)
-    print("Response status:%s reason:%s body:%s" % (
```
Comment (mbruzek) [inline hooks/hooks.py]: These values will be useful in diagnosing a problem.

File: hooks/lib/registrator.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            # go to register a new minion, we always seem to get this error.
+            # https://github.com/GoogleCloudPlatform/kubernetes/issues/1995
+            time.sleep(1)
+            print("Retrying registration...")
+            raise ValueError("Registration returned 500, retry")
+            # return register_machine(apiserver, retry=True)
+        else:
+            print("Registration error")
+            # TODO - get request data
+            raise RuntimeError("Unable to register machine with")
```
Comment (mbruzek) [inline hooks/lib/registrator.py:82]: with ... ?   Print out the response status and perhaps the result.get('message') here so the logs will tell us what is wrong.

File: hooks/lib/registrator.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        headers = {"Content-type": "application/json",
+                   "Accept": "application/json"}
+        connection = httplib.HTTPConnection(hostname, port)
+        print 'CONN {}'.format(connection)
+        connection.request("POST", api_path, json.dumps(self.data), headers)
+        response = connection.getresponse()
+        body = response.read()
+        print(body)
+        result = json.loads(body)
+        print("Response status:%s reason:%s body:%s" % (
```
Comment (lazypower) [inline hooks/lib/registrator.py:44]: And they exist right here :) They just got moved.

Comment (mbruzek) [inline hooks/lib/registrator.py:44]: Sorry I didn't see that!

File: hooks/lib/registrator.py (hunk 2 above)
Comment (lazypower) [inline hooks/lib/registrator.py:82]: Well, i noticed that the log is already spitting that information out further up in the backscroll by ~ 15 lines. I was pondering on removing the with and just leaving it be... the print statement referenced in my last diff commit has all that already. Plus we are pretty-printing the JSON object. Is it really necessary to put the failure in 3 places in the log?

Comment (mbruzek) [inline hooks/lib/registrator.py:82]: not necessary.  I didn't realize it is printed already when I made the comment. Just fix the incomplete sentence.

**Comment by mbruzek** [comment]
I deployed the develop.yaml bundle with these changes

**Pull request merged**